openmeter/billing/adapter/invoice.go

```diff
@@ -216,6 +216,17 @@ func (a *adapter) ListInvoices(ctx context.Context, input billing.ListInvoicesIn
 			))
 		}
 
+		if input.ExternalIDs != nil {
+			switch input.ExternalIDs.Type {
+			case billing.InvoicingExternalIDType:
+				query = query.Where(billinginvoice.InvoicingAppExternalIDIn(input.ExternalIDs.IDs...))
+			case billing.PaymentExternalIDType:
+				query = query.Where(billinginvoice.PaymentAppExternalIDIn(input.ExternalIDs.IDs...))
+			case billing.TaxExternalIDType:
+				query = query.Where(billinginvoice.TaxAppExternalIDIn(input.ExternalIDs.IDs...))
+			}
```

Comment:
error if unknown type?

Reply:
It's not needed as it's part of the validation: https://github.com/openmeterio/openmeter/blob/667b205098558e4fb0b5cc433892b00219ac7af4/openmeter/billing/invoice.go#L641